### PR TITLE
Fixes #15726 - properly upgrade hypervisors to facets

### PR DESCRIPTION
**Pull request by jlsherrill**
as part of content host db migration

### CODE REVIEW

File: db/migrate/20150930183738_migrate_content_hosts.rb [hunk 1]
```diff
@@ -295,6 +297,37 @@ def unregister_all_but_last_system(systems)
...
+      name = "virt-who-#{hypervisor.name}-#{org.id}"
+
+      host = ::Host.find_by(:name => name)
+      if host.nil?
+        host = MigrateContentHosts::Host.create!(:name => name, :organization => org.becomes(MigrateContentHosts::Organization), :managed => false,
+                                             :type => "Host::Managed", :location => MigrateContentHosts::Location.default_location)
+      elsif host.organization != org
+        logger.warn("Found host with name #{name} but it's in org #{host.organization.name} instead of #{org.name}.")
+        unregister_system(hypervisor)
+        next
```
Comment (ehelms) [inline db/migrate/20150930183738_migrate_content_hosts.rb]: :+1:

**Comment by jlsherrill** [comment]
My virt-who config:

/etc/virt-who.d/test.conf

[libvirt]
type=libvirt
server=qemu:///system
owner=Default_Organization
env=Library
hypervisor_id=hostname
rhsm_hostname=katello.example.com
rhsm_username=admin
rhsm_password=changeme
rhsm_prefix=/rhsm

File: db/migrate/20150930183738_migrate_content_hosts.rb [hunk 2]
```diff
@@ -295,6 +297,37 @@ def unregister_all_but_last_system(systems)
...
+  end
+
+  def update_hypervisor_consumer(hypervisor)
+    org = hypervisor.environment.organization
+    hypervisor.content_view = org.default_content_view.becomes(MigrateContentHosts::ContentView)
+    hypervisor.environment = org.library.becomes(MigrateContentHosts::KTEnvironment)
+    hypervisor.save!
+
+    candlepin_environment_id = org.default_content_view.content_view_environments.first.cp_id
+    ::Katello::Resources::Candlepin::Consumer.update(hypervisor.uuid, 'environment' => {'id' => candlepin_environment_id})
```
Comment (ehelms) [inline db/migrate/20150930183738_migrate_content_hosts.rb]: Previously this migration was only reading data, and the rake task handled updates, this introduces doing updates now in the migration?

Comment (jlsherrill) [inline db/migrate/20150930183738_migrate_content_hosts.rb]: Good point and correct.  It seems simpler to do it here and re-doing this action multiple times is not harmful in any way.  

I can't think of any advantages to splitting it out to a rake task other than to follow this 'rule' that this task has already broken :/   I'm not opposed to doing it just not sure of the benefit at this point.

**Comment by ehelms** [comment]
Worked as expected for me - ACK

**Pull request merged**